components/eamxx/src/share/grid/remap/vertical_remapper.hpp

```diff
@@ -124,13 +126,25 @@ class VerticalRemapper : public AbstractRemapper
   std::vector<Field>    m_src_masks;
 
   // Vertical profile fields, both for source and target
-  int                   m_num_remap_levs;
   Real                  m_mask_val;
-  Field                 m_remap_pres;
-  Field                 m_src_mid;  // Src vertical profile for LEV layouts
-  Field                 m_src_int;  // Src vertical profile for ILEV layouts
-  bool                  m_mid_set = false;
-  bool                  m_int_set = false;
+  Field                 m_tgt_pressure;
+  Field                 m_src_pmid;  // Src vertical profile for LEV layouts
+  Field                 m_src_pint;  // Src vertical profile for ILEV layouts
+
+  // We need to remap mid/int fields separately, and we want to use packs if possible,
+  // so we need to divide input fields into 4 separate categories
+
+  // Map field id to whether it's packed/scalar and midpoint/interface
+  struct FType {
+    bool packed = false;
+    bool midpoints = true;
+  };
+  std::map<std::string,FType> m_field2type;
+
+  std::shared_ptr<ekat::LinInterp<Real,SCREAM_PACK_SIZE>> m_lin_interp_mid_packed;
+  std::shared_ptr<ekat::LinInterp<Real,SCREAM_PACK_SIZE>> m_lin_interp_int_packed;
+  std::shared_ptr<ekat::LinInterp<Real,1>>                m_lin_interp_mid_scalar;
+  std::shared_ptr<ekat::LinInterp<Real,1>>                m_lin_interp_int_scalar;
```

Comment:
It may not be too important, but this is another time where it would be nice to be able to have a scalar version (and not just pack size 1). Should be a quick task, may look into this over leave if you haven't already.

Reply:
Yes, it would be nice. But then you'd have 2 impls for LinInterp. I think at some point we need to a) think about switching to Kokkos::simd, and b) go through our basic functions, and make sure we can call them with either packs or scalars. E.g., index_and_shift (whose name I would probably change) only works for packed views, forcing different impls. If we had something like
```
template<typename ScalarT, typename IndexT>
ScalarT eval_at_index (const view_1d<ScalarT>& v, const IndexT& i);
```
with specializations for Packs[^1], then we may be able to have a single impl for LinInterp, which works for both packs and scalars.

[^1]: since partial specialization is not possible for function, we'd need a helper struct, but that's a minor detail.